language/stdlib/modules/offer.mvir

```diff
@@ -0,0 +1,40 @@
+// TODO: add optional timeout for reclaiming by original publisher once we have implemented time
+module Offer {
+    // A wrapper around value `offered` that can be claimed by the address stored in `for`.
+    resource T<Offered> { offered: Offered, for: address }
+
+    // Publish a value of type `Offered` under the sender's account. The value can be claimed by
+    // either the `for` address or the transaction sender.
+    public create<Offered>(offered: Offered, for: address) {
+        move_to_sender<T<Offered>>(T<Offered> { offered: move(offered), for: move(for) });
+        return;
+    }
+
+    // Claim the value of type `Offered` published at `offer_address`.
+    // Only succeeds if the sender is the intended recipient stored in `for` or the original
+    // publisher `offer_address`.
```

Comment:
Great design! I guess this is used as a cancellation safety net?

Reply:
Correct :). Once we have on-chain time, I think it might make sense to only let the publisher reclaim it after a timeout.